okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/LeakDetector.java

```diff
@@ -108,4 +101,15 @@ private void logLeak(Optional<RuntimeException> stackTrace) {
             this.stackTrace = stackTrace;
         }
     }
+
+    static boolean isTraceLoggingEnabled() {
+        return log.isTraceEnabled();
+    }
+
+    interface Reporter {
```

Comment:
I was a a little confused by why this Reporter interface is actually necessary - this whole LeakDetector thing is only ever used in the ConcurrencyLimiters class, so defining an inline anonymous implementation there feels a little weird.

What about moving the implementation into this file, or even squashing some of the logic into this actual class?